### PR TITLE
[Bugfix]UD resolved address / got callback

### PR DESCRIPTION
**Pull request by kcw-grunt**
## Issue
The UD resolution was inconsistent before (60% of the time)

## Approach
Looked into the steps looking for the callback 

## Commits
- Added a dispatch group
- Fixed the bug by  adding wait to the group from (@JohnnyJumper)
- Refactored partner plist

### CODE REVIEW

File: loafwallet/UnstoppableDomainViewModel.swift [hunk 1]
```diff
@@ -72,40 +70,92 @@ class UnstoppableDomainViewModel: ObservableObject {
...
+
+        } else {
+            print("Error: resolution is not setup")
+        }
+    }
+}
+
 
-                    }
+class ResolutionModel: NSObject {
```
Comment (mosadialiou) [inline loafwallet/UnstoppableDomainViewModel.swift:134]: you have to delete this class. it's not used!

File: loafwallet/UnstoppableDomainViewModel.swift [hunk 2]
```diff
@@ -72,40 +70,92 @@ class UnstoppableDomainViewModel: ObservableObject {
...
-                            LWAnalytics.logEventWithParameters(itemName:
-                                                                CustomEvent._20201121_FRIA,
-                                                               properties:
-                                                                ["failure_time": timestamp,
-                                                                 "error":error.localizedDescription])
-                            
-                            print("Expected LTC Address, but got \(error.localizedDescription)")
+        // This group is created to allow the threads to complete.
+        // Otherwise, we may never get in the callback relative to UDR v0.1.6
+        let group = DispatchGroup()
```
Comment (mosadialiou) [inline loafwallet/UnstoppableDomainViewModel.swift:75]: can you extract a method?

File: loafwallet/ResolutionModel.swift [hunk 3]
```diff
@@ -9,30 +9,30 @@
 import Foundation
```
Comment (mosadialiou) [inline loafwallet/ResolutionModel.swift:9]: we should remove this class, right?

**Review by mosadialiou — CHANGES_REQUESTED** [review]

File: loafwallet/ResolutionModel.swift (hunk 3 above)
Comment (kcw-grunt) [reply loafwallet/ResolutionModel.swift:9]: my bad...removing

File: loafwallet/UnstoppableDomainViewModel.swift (hunk 2 above)
Comment (kcw-grunt) [reply loafwallet/UnstoppableDomainViewModel.swift:75]: Sorry...I'm not understanding

Comment (mosadialiou) [reply loafwallet/UnstoppableDomainViewModel.swift:75]: create a function or method to export all this new code

Comment (mosadialiou) [reply loafwallet/UnstoppableDomainViewModel.swift:75]: you can call the function or method `resolveDomain`

Comment (kcw-grunt) [reply loafwallet/UnstoppableDomainViewModel.swift:75]: yeah in troubleshooting I put it all there. Let me move it.

File: loafwallet/UnstoppableDomainViewModel.swift [hunk 4]
```diff
@@ -10,27 +10,36 @@ import Foundation
 import SwiftUI
 import Combine
 import UnstoppableDomainsResolution
- 
+
+public typealias KeyPathResult = (Result<String, DocumentError>) -> Void
```
Comment (mosadialiou) [inline loafwallet/UnstoppableDomainViewModel.swift:14]: where do you use `KeyPathResult`?

**Review by mosadialiou — CHANGES_REQUESTED** [review]

File: loafwallet/UnstoppableDomainViewModel.swift (hunk 4 above)
Comment (kcw-grunt) [reply loafwallet/UnstoppableDomainViewModel.swift]: removed (as well as the DocumentError enum)

File: loafwallet/UnstoppableDomainViewModel.swift [hunk 5]
```diff
@@ -10,27 +10,29 @@ import Foundation
...
     //MARK: - Public Variables
     var didResolveUDAddress: ((String) -> Void)?
-     
+    
     var shouldClearAddressField: (() -> Void)?
     
     //MARK: - Private Variables
     private var ltcAddress = ""
     
+    private var resolution: Resolution?
```
Comment (mosadialiou) [inline loafwallet/UnstoppableDomainViewModel.swift:34]: not used! delete it and we are good

**Review by mosadialiou — CHANGES_REQUESTED** [review]

File: loafwallet/UnstoppableDomainViewModel.swift (hunk 5 above)
Comment (kcw-grunt) [reply loafwallet/UnstoppableDomainViewModel.swift:34]: ugh..caught me!

**Review by mosadialiou — APPROVED** [review]
LGTM

**Pull request merged**